sql/2025/05/featured-image-block-usage.sql

```diff
@@ -0,0 +1,73 @@
+# HTTP Archive query to get % WordPress sites using Featured Image in block themes.
+#
+# WPP Research, Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/184
+
+DECLARE
+  DATE_TO_QUERY DATE DEFAULT '2025-03-01';
+
+CREATE TEMPORARY FUNCTION IS_CMS(technologies ARRAY<STRUCT<technology STRING, categories ARRAY<STRING>, info ARRAY<STRING>>>, cms STRING, version STRING) RETURNS BOOL AS (
+  EXISTS(
+    SELECT * FROM UNNEST(technologies) AS technology, UNNEST(technology.info) AS info
+    WHERE technology.technology = cms
+    AND (
+      version = ""
+      OR ENDS_WITH(version, ".x") AND (STARTS_WITH(info, RTRIM(version, "x")) OR info = RTRIM(version, ".x"))
+      OR info = version
+    )
+  )
+);
+
+WITH wordpress AS (
+  SELECT
+    client,
+    page
+  FROM
+    `httparchive.crawl.pages`
+  WHERE
+    date = DATE_TO_QUERY
+    AND IS_CMS(technologies, 'WordPress', '5.9.0')
+    AND is_root_page = TRUE
+),
+
+featuredImageBlockDetection AS (
+  SELECT
+    client,
+    page,
+    REGEXP_CONTAINS(response_body, r'<figure class="wp-block-post-featured-image">') AS has_featured_image_block
+  FROM
+    `httparchive.crawl.requests`
+  WHERE
+    date = DATE_TO_QUERY
+    AND is_root_page = TRUE
+    AND is_main_document
+)
```

Comment:
The query right now looks as featured image block across _all WordPress sites_, which is a low number (~1.2%). What's missing is filtering by block themes.

```suggestion
featuredImageBlockDetection AS (
  SELECT
    client,
    page,
    REGEXP_CONTAINS(response_body, r'<figure class="wp-block-post-featured-image">') AS has_featured_image_block
  FROM
    `httparchive.crawl.requests`
  WHERE
    date = DATE_TO_QUERY
    AND is_root_page = TRUE
    AND is_main_document
    AND response_body LIKE '%<div class="wp-site-blocks">%'
)
```

With this, you'll get more reasonable results:

```

desktop | 19820 | 53448 | 0.37082772040113754 |  
-- | -- | -- | -- | --
2 | mobile | 25700 | 68758 | 0.37377468803630121 |  

```

In other words: 37% of all WordPress sites using a block theme use the featured image block on the homepage.

Reply:
Thanks for sharing the results. Based on the data, we should prioritize updating it to ensure accurate sizes, it's a great opportunity since 37% is a significant portion of sites.